templates/sqoop-parquet-hdfs-hive-merge/hdfs-incr-clear.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+{#  Copyright 2017 Cargill Incorporated
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License. #}
+
+# Remove parquet data from incr
+set -eu
+sudo -u hdfs hdfs dfs -rm -r -f {{ conf.staging_database.path }}/{{ table.destination.name }}/incr/*
```

Comment:
Do you need to run this command as the hdfs user? Most people won't have the ability to run sudo commands in their pipelines. It's dangerous if you aren't in a docker container.

Reply:
I agree but I had to. Without sudo, it can get into a situation where one cannot write due to:
`WARNINGS: Impala does not have READ_WRITE access to path 'hdfs://0.0.0.0:8020/user/hive/warehouse/titanic'`
While one can clearly add the write via chmod:
`merge/titanic# hadoop fs -ls /user/hive/warehouse/titanic`
`Found 2 items`
`drwxrwxrwx   - root supergroup          0 2018-04-10 19:20 /user/hive/warehouse/titanic/base`
`drwxrwxrwx   - root supergroup          0 2018-04-10 19:20 /user/hive/warehouse/titanic/incr`

I feel I'm missing something basic here wrt access control in docker.

Reply:
We need to get to the root of this problem. The other calls to sudo are part of the integration test, and the user has sudo during the integration tests in Docker.

This command though will be run by engineers that don't have sudo.

I think this might be related to https://github.com/Cargill/pipewrench/issues/36

Try running your test before `$SCRIPT_DIR/sqoop-parquet-hdfs-kudu-impala/run.sh`